src/FSLibrary/MaxTPSTest.fs

```diff
@@ -132,35 +132,110 @@ let maxTPSTest (context: MissionContext) (baseLoadGen: LoadGen) (setupCfg: LoadG
                 context.image
                 (if context.flatQuorum.IsSome then context.flatQuorum.Value else false)
 
+    // PayPregenerated requires node restart between failed iterations to ensure validity of the pregenerated transactions
+    // However, large-scale simulation restarts can be slow, so for now only use the new mode on small networks
+    let lgMode = if List.length allNodes > 30 then GeneratePaymentLoad else PayPregenerated
+    let baseLoadGen = { baseLoadGen with mode = lgMode }
+
+    let context =
+        { context with
+              genesisTestAccountCount =
+                  if baseLoadGen.mode = PayPregenerated then
+                      Some(context.genesisTestAccountCount |> Option.defaultValue 500000)
+                  else
+                      None
+              numPregeneratedTxs =
```

Comment:
Do you know what TPS this number of txs gets us? It might need to eventually be dynamic based on the max TPS we test

Reply:
well the test runs for 1000 seconds, so 500000 / 1000 = 500 tps per loadgen node. For now it's plenty of headroom, I agree in the future we can make this dynamic though.

Reply:
I also don't necessarily know if it makes sense to go above 500 TPS per node. While loadgen is much cheaper now, it's still not free, so at that point we'll be adding quite a bit of overhead anyways.